lib/interfaces.ts

```diff
@@ -0,0 +1,66 @@
+import { ComputedRef, DeepReadonly, Ref, UnwrapNestedRefs, WatchCallback } from 'vue'
+
+export interface IModule<
+  S extends IState,
+  G extends IGetters,
+  M extends IMutations,
+  SM extends ISubModules = never,
+> {
+  readonly options: IModuleOptions<S, G, M, SM>
+  flatten(): IFlattenedModule<S, G, M, SM>
```

Comment:
I'm still not sure if we should require the end user to call `flatten`, or if it should be done automatically, and only the flattened module type is exposed.